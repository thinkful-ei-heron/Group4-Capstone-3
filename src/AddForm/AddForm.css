.add-form-page h2 {
	color: black;
}
.add-form-page nav input {
	display: none;
}

.add-form-page form {
	background-image: url("../images/wall6.jpg");
	background-repeat: repeat;
	background-size: ;
	margin-right: 0em;
	padding: 49px;
}

.dashboard-top {
	background-image: url('../assets/img/hops5.jpg');
	background-size: cover;
	background-position: center;
	background-repeat: no-repeat;
}

.add-form-page section.header-component {
	margin: 0px;
}

.add-form-page nav {
	background-color: rgba(8, 8, 8);
	height: 2.3em;
	text-align: left;
}
.add-form-page nav a:hover {
	color: rgb(8, 184, 8);
}
.add-form-page nav button {
	float: right;
	color: white;
	font-family: 'Merriweather', serif;
	font-size: 1.5em;
	border: none;
	height: 1.3em;
}
.add-form-page nav button:hover {
	color: rgb(8, 184, 8);
}
.add-form-page h3 {
	font-family: 'Roboto Slab', serif;
	color: white;
	font-size: 25px;
}

.add-form-page input,
textarea {
	border-radius: 3px;
	-webkit-border-radius: 3px;
	-moz-border-radius: 3px;
	border: 1px solid rgb(8, 184, 8);
	outline: none;
	color: rgb(82, 79, 80);
	padding: 5px 8px 5px 8px;
	box-shadow: inset 1px 1px 4px rgb(198, 189, 192);
	-moz-box-shadow: inset 1px 1px 4px rgb(158, 149, 153);
	-webkit-box-shadow: inset 1px 1px 4px rgb(110, 107, 108);
	background: rgb(178, 220, 183);
}
.clear-submit-back-btn {
	background: rgb(63, 62, 63);
	border: 1px solid rgb(27, 10, 18);
	padding: 5px 15px 5px 15px;
	color: rgb(122, 194, 125);
	box-shadow: inset -1px -1px 3px rgb(65, 63, 64);
	-moz-box-shadow: inset -1px -1px 3px rgb(100, 96, 98);
	-webkit-box-shadow: inset -1px -1px 3px rgb(105, 101, 103);
	border-radius: 3px;
	border-radius: 3px;
	-webkit-border-radius: 3px;
	-moz-border-radius: 3px;
	font-weight: bold;
	margin: 10px;
}

.add-form-page fieldset {
	width: 50%;
	margin-top: 20px;
	border-radius: 10px;
	border: 2px solid rgba(60, 131, 64, 0.858);
	padding: 20px;
	box-shadow: inset 0px 0px 15px rgb(243, 230, 230);
	-moz-box-shadow: inset 0px 0px 15px rgb(236, 217, 217);
	-webkit-box-shadow: inset 0px 0px 15px rgb(215, 209, 209);
	background-color: rgba(255, 255, 255, 0.66);
	z-index: 1;
}
.unstyled::-webkit-inner-spin-button {
	display: none;
	-webkit-appearance: none;
}

.add-form-page textarea {
	width: 90%;
	height: 100px;
}
.form-section {
	padding: 10px;
	display: flex;
	flex-direction: row;
	justify-content: center;
}

.form-section label {
	color: black;
	font-weight: bold;
	font-size: 15px;
	text-shadow: 1px 1px 1px #fff;
	margin: 0px 0px 20px 15px;
}

#beer-entry-abv,
#beer-entry-location,
#beer-entry-name,
#beer-entry-date {
	width: 90%;
	color: rgb(80, 70, 74);
	font-weight: bold;
	font-size: 13px;
	text-shadow: 1px 1px 1px #fff;
}

#beer-entry-type {
	width: 90%;
	background: rgb(63, 62, 63);
	border: 1px solid rgb(27, 10, 18);
	padding: 5px 15px 5px 15px;
	color: rgb(122, 194, 125);
	font-size: 15px;
}

.add-form-page legend {
	color: rgb(21, 53, 30);
	border-top: 2px solid rgb(73, 159, 70);
	border-left: 2px solid rgb(73, 159, 70);
	border-right: 2px solid rgb(95, 155, 100);
	border-radius: 5px 5px 0px 0px;
	-webkit-border-radius: 5px 5px 0px 0px;
	-moz-border-radius: 5px 5px 0px 0px;
	background: rgba(226, 240, 227, 0.653);
	padding: 0px 8px 3px 8px;
	box-shadow: -0px -1px 2px #f1f1f1;
	-moz-box-shadow: -0px -1px 2px #f1f1f1;
	-webkit-box-shadow: -0px -1px 2px #f1f1f1;
	font-weight: normal;
	font-size: 20px;
}

.required {
	color: red;
	font-weight: normal;
	position: relative;
}
.add-form-page div {
	margin: 10px;
}

#beer-entry-description label,
textarea {
	position: relative;
	top: 20px;
}

@media only screen and (max-width: 570px) {
	.form-section {
		padding: 10px;
		display: flex;
		flex-direction: column;
		justify-content: center;
	}
	.add-form-page input:checked ~ a {
		margin-bottom: 0px;
	}

	.add-form-page nav {
		background-color: transparent;
	}
	.add-form-page .header-component a {
		font: 10px;
		margin: 0px;
	}
	.add-form-page nav input {
		display: inline-block;
		width: 60px;
		height: 50px;
		position: absolute;
		top: 123px;
		left: 2px;
		cursor: pointer;
		opacity: 0;
		-webkit-touch-callout: none;
	}
	.add-form-page nav span {
		display: block;
		width: 33px;
		height: 4px;
		margin-bottom: 5px;
		position: relative;
		background: white;
		border-radius: 3px;
		z-index: 1;
		transform-origin: 4px 0px;
		transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0),
			background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0), opacity 0.55s ease;
	}

	.add-form-page a {
		padding-top: 0px;
		margin-left: 0px;
		display: block;
	}
	.add-form-page nav span:first-child {
		transform-origin: 0% 0%;
	}

	.add-form-page nav input:checked ~ .child-2 {
		position: relative;
		top: 15px;
		transform-origin: 0% 100%;
		transform: rotate(-20deg) translate(2px, -1px);
	}

	.add-form-page a.active {
		border-bottom: none;
	}

	.add-form-page input:checked ~ span {
		opacity: 1;
		transform: rotate(45deg) translate(-2px, -1px);
		background: white;
	}

	.add-form-page nav input:checked ~ .child-3 {
		opacity: 0;
		transform: rotate(0deg) scale(0.2, 0.2);
	}

	.add-form-page nav input:checked ~ .child-1 {
		transform: rotate(20deg) translate(0, -1px);
		position: relative;
		top: 15px;
	}

	.add-form-page nav a,
	a.active {
		color: black;
		font-weight: bold;
		margin-left: 0px;
		list-style-type: none;
		-webkit-font-smoothing: antialiased;
		transform-origin: 0% 0%;
		transform: translate(-100%, 0);
		transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1.0);
		width: 200px;
	}

	.add-form-page nav input:checked ~ a {
		transform: none;
		display: flex;
		position: relative;
		top: 12px;
		left: -7px;
		margin-right: 10px;
		background-color: rgba(226, 240, 227, 0.653);
		border: 1px solid rgba(60, 131, 64, 0.858);
		border-radius: 5px;
		font-size: 15px;
		margin-left: 12px;
		padding: 3px;
	}
	.add-form-page .header-component nav button {
		color: white;
		display: block;
		float: right;
		position: relative;
		top: -111px;
		right: 148px;
		font-weight: bold;
		font-size: 20px;
	}

	.add-form-page nav {
		background-color: rgba(8, 8, 8, 0.774);
		height: 2.3em;
		text-align: left;
	}
}
