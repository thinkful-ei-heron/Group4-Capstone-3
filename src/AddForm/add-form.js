import React, {Component} from 'react';
import JournalContext from "../contexts/JournalContext";
import BeerApiService from "../services/beer-api-service";
import NavBar from "../NavBar/NavBar"

export default class AddForm extends Component {

    static contextType = JournalContext;

    state = {error: null};

    firstInput = React.createRef();

    handleSubmit = ev => {
        ev.preventDefault();
        const {username, password} = ev.target;

        this.setState({error: null});

        BeerApiService.postBeer({
            user_name: username.value,
            password: password.value,
        })
            .then(res => {
                username.value = "";
                password.value = "";
                this.context.processLogin(res.authToken);
                this.handleLoginSuccess();
            })
            .catch(res => {
                this.setState({error: res.error});
            });
    };

    render() {
        const {error} = this.state;
        return (
            <div>
                <h1>The Dear Beer</h1>
                <NavBar/>
                <h3>Add Beer Entry</h3>
                <form onSubmit={this.handleSubmit}>
                    <div role='alert'>
                        {error && <p className='error'>Something went wrong!</p>}
                    </div>
                    <section>
                        <div>
                            <label htmlFor='beer-entry-date'>Date</label>
                            <input type='text' id='beer-entry-date'  value={this.context.date_created}
                                   onChange={(e)=>this.context.setDate(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-name'>Name</label>
                            <input type='text' id='beer-entry-name' value={this.context.name}
                                   onChange={(e)=>this.context.setName(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-location'>Loc</label>
                            <input type='text' id='beer-entry-location' value={this.context.location}
                                   onChange={(e)=>this.context.setLoc(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-rating'>Rating</label>
                            <input type="range" min="0" max="9" value={this.context.rating}
                                   onChange={(e)=>this.context.setRating(e.target.value)} className="slider"
                                   id="rating" step="1"/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-apv'>APV</label>
                            <input type='number' id='beer-entry-apv' value={this.context.apv} onChange={(e)=> this.context.setApv(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-rating'>Description</label>
                            <textarea id='beer-entry-description' value={this.context.description} onChange={(e)=> this.context.setDesc(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-color'>Dark to Light</label>
                            <input type="range" id="beer-entry-color" min="1" max="6" step='1' value={this.context.color}  onChange={(e)=> this.context.setColor(e.target.value)} />
                        </div>
                        <div>
                            <label htmlFor='beer-entry-heaviness'>Heavy to Light</label>
                            <input type="range" min="1" max="5" className="slider" id="heaviness" step='1' value={this.context.heaviness}  onChange={(e)=> this.context.setHeaviness(e.target.value)}/>
                        </div>
                        <div>
                            <button type='button' onClick={this.context.resetAll}>Clear</button>
                            <button type='submit'>Submit</button>
                        </div>
                    </section>
                </form>
            </div>
        )
    }
}