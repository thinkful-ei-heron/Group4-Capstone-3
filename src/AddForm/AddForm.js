import React, {Component} from 'react';
import JournalContext from "../contexts/JournalContext";
import BeerApiService from "../services/beer-api-service";
import Header from "../Header/Header"
import Utils from '../Utils/Utils'
import './AddForm.css' 

export default class AddForm extends Component {

    static contextType = JournalContext;

    state = {error: null, color: 1};

    firstInput = React.createRef();


    componentDidMount() {
        this.setState({color: this.context.color})
    }

    componentDidUpdate(prevProps, prevState, snapshot) {
        if (prevState.color !== this.state.color) {
            this.setState({color: this.context.color})
        }
    }


    handleSubmit = ev => {
        ev.preventDefault();
        //const checkedValues = this.context.checkValues();

        // if(!checkedValues.bool)
        //     this.setState({error: checkedValues.error});
        // else {
        BeerApiService.postBeer({
            name: this.context.name,
            date_created: this.context.date_created,
            location: this.context.location,
            description: this.context.description,
            type: this.context.type,
            rating: this.context.rating,
            abv: this.context.abv,
            heaviness: this.context.heaviness,
            color: this.context.color
        })
            .then(res => {
                this.context.resetAll();
                window.location.replace('/home');
            })
            .catch(res => {
                this.setState({error: res.error});
            });
    };

    render() {
        const {error} = this.state;
        return (
            <div className='add-form-page'>
                <Header location={this.props.location} header={'Add Beer'}/>
                <form onSubmit={this.handleSubmit}>
                    <div role='alert'>
                        {error && <p className='error'>Something went wrong!</p>}
                    </div>
                    <section >
                        <figure>
                            <img src={Utils.getImage(this.context.color)} alt={'Beer'} width={'150px'}/>
                        </figure>

                        <div>
                            <label htmlFor='beer-entry-date'>Date</label>
                            <input type='date' id='beer-entry-date' value={this.context.date_created}
                                   onChange={(e) => this.context.setDate(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-name'>Name</label>
                            <input type='text' id='beer-entry-name' value={this.context.name}
                                   onChange={(e) => this.context.setName(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-location'>Location</label>
                            <input type='text' id='beer-entry-location' value={this.context.location}
                                   onChange={(e) => this.context.setLoc(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-rating'>Rating</label>
                            <input type="range" min="0" max="9" value={this.context.rating}
                                   onChange={(e) => this.context.setRating(e.target.value)} className="slider"
                                   id="rating" step="1"/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-abv'>ABV</label>
                            <input type='number' id='beer-entry-abv' value={this.context.abv}
                                   onChange={(e) => this.context.setAbv(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-type'>Type</label>
                            <input type='text' id='beer-entry-type' value={this.context.type}
                                   onChange={(e) => this.context.setType(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-rating'>Description</label>
                            <textarea id='beer-entry-description' value={this.context.description}
                                      onChange={(e) => this.context.setDesc(e.target.value)} required/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-color'>Dark to Light</label>
                            <input type="range" id="beer-entry-color" min="1" max="6" step='1'
                                   value={this.context.color}
                                   onChange={(e) => {
                                       this.context.setColor(parseInt(e.target.value));
                                       this.setState({color: parseInt(e.target.value)})
                                   }}/>
                        </div>
                        <div>
                            <label htmlFor='beer-entry-heaviness'>Heavy to Light</label>
                            <input type="range" min="1" max="5" className="slider" id="heaviness" step='1'
                                   value={this.context.heaviness}
                                   onChange={(e) => this.context.setHeaviness(e.target.value)}/>
                        </div>
                        <div>
                            <button type='button' onClick={this.context.resetAll}>Clear</button>
                            <button type='submit'>Submit</button>
                        </div>
                    </section>
                </form>
            </div>
        
        )
    }
}