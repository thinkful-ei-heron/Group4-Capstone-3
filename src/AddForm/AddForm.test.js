import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AddForm from './AddForm'

describe(`AddForm component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop'
      }

  it('renders the complete form', () => {
    const wrapper = shallow(<AddForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the EmmaForm given props', () => {
    const wrapper = shallow(<AddForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})