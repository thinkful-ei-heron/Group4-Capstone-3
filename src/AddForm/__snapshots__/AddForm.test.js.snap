// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm component renders the Form given props 1`] = `
<div
  className="add-form-page"
>
  <Header
    header="Add Beer"
  />
  <form
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <section
      className="form-section"
    >
      <fieldset>
        <legend>
          Beer Information
        </legend>
        <figure>
          <img
            alt="Beer"
            src="light.jpg"
            width="150px"
          />
        </figure>
        <div>
          <label
            htmlFor="beer-entry-date"
          >
            Date
          </label>
          <input
            className="unstyled"
            id="beer-entry-date"
            onChange={[Function]}
            type="date"
          />
        </div>
        <div>
          <label
            htmlFor="beer-entry-location"
          >
            Location
          </label>
          <span
            className="required"
          >
            *
          </span>
          <input
            id="beer-entry-location"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="beer-entry-name"
          >
            Name
          </label>
          <span
            className="required"
          >
            *
          </span>
          <input
            id="beer-entry-name"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div>
          <label
            htmlFor="beer-entry-type"
          >
            Type
          </label>
          <span
            className="required"
          >
            *
          </span>
          <select
            id="beer-entry-type"
            onChange={[Function]}
            required={true}
          >
            <option
              key="0"
              value="Ale"
            >
              Ale
            </option>
            <option
              key="1"
              value="Altbier"
            >
              Altbier
            </option>
            <option
              key="2"
              value="American Lager"
            >
              American Lager
            </option>
            <option
              key="3"
              value="Barley Wine"
            >
              Barley Wine
            </option>
            <option
              key="4"
              value="Belgian"
            >
              Belgian
            </option>
            <option
              key="5"
              value="Berliner Weisse"
            >
              Berliner Weisse
            </option>
            <option
              key="6"
              value="Bitter"
            >
              Bitter
            </option>
            <option
              key="7"
              value="Bock"
            >
              Bock
            </option>
            <option
              key="8"
              value="Brown Ale"
            >
              Brown Ale
            </option>
            <option
              key="9"
              value="Brown Ale"
            >
              Brown Ale
            </option>
            <option
              key="10"
              value="Cider"
            >
              Cider
            </option>
            <option
              key="11"
              value="Cream Ale"
            >
              Cream Ale
            </option>
            <option
              key="12"
              value="Doppelbock"
            >
              Doppelbock
            </option>
            <option
              key="13"
              value="Dunkel"
            >
              Dunkel
            </option>
            <option
              key="14"
              value="Flanders Red Ale"
            >
              Flanders Red Ale
            </option>
            <option
              key="15"
              value="German Pilser"
            >
              German Pilser
            </option>
            <option
              key="16"
              value="Gose"
            >
              Gose
            </option>
            <option
              key="17"
              value="Helles"
            >
              Helles
            </option>
            <option
              key="18"
              value="Helles Bock"
            >
              Helles Bock
            </option>
            <option
              key="19"
              value="Honey"
            >
              Honey
            </option>
            <option
              key="20"
              value="Imperial IPA"
            >
              Imperial IPA
            </option>
            <option
              key="21"
              value="IPA"
            >
              IPA
            </option>
            <option
              key="22"
              value="Irish Red Ale"
            >
              Irish Red Ale
            </option>
            <option
              key="23"
              value="Kolsch"
            >
              Kolsch
            </option>
            <option
              key="24"
              value="Lager"
            >
              Lager
            </option>
            <option
              key="25"
              value="Lambic"
            >
              Lambic
            </option>
            <option
              key="26"
              value="Mild Ale"
            >
              Mild Ale
            </option>
            <option
              key="27"
              value="Old Ale"
            >
              Old Ale
            </option>
            <option
              key="28"
              value="Pale Ale"
            >
              Pale Ale
            </option>
            <option
              key="29"
              value="Pale Lager"
            >
              Pale Lager
            </option>
            <option
              key="30"
              value="Pilsner"
            >
              Pilsner
            </option>
            <option
              key="31"
              value="Porter"
            >
              Porter
            </option>
            <option
              key="32"
              value="Quadrupel"
            >
              Quadrupel
            </option>
            <option
              key="33"
              value="Rye"
            >
              Rye
            </option>
            <option
              key="34"
              value="Saison"
            >
              Saison
            </option>
            <option
              key="35"
              value="Schwarzbier"
            >
              Schwarzbier
            </option>
            <option
              key="36"
              value="Scotch Ale"
            >
              Scotch Ale
            </option>
            <option
              key="37"
              value="Seasonal Beer"
            >
              Seasonal Beer
            </option>
            <option
              key="38"
              value="Stout"
            >
              Stout
            </option>
            <option
              key="39"
              value="Vienna Lager"
            >
              Vienna Lager
            </option>
            <option
              key="40"
              value="Wittbier"
            >
              Wittbier
            </option>
            <option
              key="41"
              value="Other"
            >
              Other
            </option>
          </select>
        </div>
        <div>
          <label
            data-tip="Alcohol By Volume"
            htmlFor="beer-entry-abv"
          >
            ABV
          </label>
          <span
            className="required"
          >
            *
          </span>
          <ReactTooltip
            clickable={false}
            effect="solid"
            insecure={true}
            place="top"
            resizeHide={true}
            type="dark"
            wrapper="div"
          />
          <input
            id="beer-entry-abv"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div>
          <label
            data-tip="Color of beer"
            htmlFor="beer-entry-color"
          >
            Dark to Light
          </label>
          <span
            className="required"
          >
            *
          </span>
          <input
            id="beer-entry-color"
            max="6"
            min="1"
            onChange={[Function]}
            step="1"
            type="range"
          />
        </div>
        <div>
          <label
            data-tip="Strength of alcohol"
            htmlFor="beer-entry-heaviness"
          >
            Heavy to Light
          </label>
          <span
            className="required"
          >
            *
          </span>
          <input
            className="slider"
            id="heaviness"
            max="5"
            min="1"
            onChange={[Function]}
            step="1"
            type="range"
          />
        </div>
      </fieldset>
      <fieldset
        className="lastfield"
      >
        <legend>
          What I think about this beer!
        </legend>
        <div>
          <img
            alt="Rating icons"
            className="rating-img"
            height="50px"
            src="half.png"
          />
          <div>
            <label
              htmlFor="beer-entry-rating"
            >
              Rating
            </label>
            <span
              className="required"
            >
              *
            </span>
            <input
              className="slider"
              id="rating"
              max="9"
              min="0"
              onChange={[Function]}
              step="1"
              type="range"
            />
          </div>
        </div>
        <label
          className="desc-label"
          htmlFor="beer-entry-description"
        >
          Description
        </label>
        <span
          className="required"
        >
          *
        </span>
        <textarea
          id="beer-entry-description"
          onChange={[Function]}
          required={true}
        />
        <button
          className="clear-submit-back-btn submit-btn"
          type="submit"
        >
          Submit
        </button>
        <div
          className="btns-2"
        >
          <Link
            to="/home"
          >
            <button
              className="clear-submit-back-btn"
              type="button"
            >
              Back
            </button>
          </Link>
          <button
            className="clear-submit-back-btn"
            type="button"
          >
            Clear
          </button>
        </div>
      </fieldset>
    </section>
  </form>
</div>
`;

exports[`AddForm component renders the complete form 1`] = `
<ContextProvider
  value={
    ParallaxController {
      "createElement": [Function],
      "destroy": [Function],
      "getElements": [Function],
      "removeElementById": [Function],
      "resetElementStyles": [Function],
      "update": [Function],
      "updateElementPropsById": [Function],
      "updateScrollContainer": [Function],
    }
  }
>
  <AddForm />
</ContextProvider>
`;
